### ISSUE TITLE
ASMu2D::evalPoint does not work

### ISSUE THREAD

**Issue by kmokstad** [OPM/IFEM#217]
The current implementation is no good, it gives the correct Cartesian coordinates yes, but it also needs to return the index of the matching nodal (control) point, if any. This is needed, e.g., for running adaptive simulations of plate problems with point loads.

In the first glance I thought it should be as easy as checking which of the `fe.N` values is equal to 1, but it seems like the `LRSplineSurface::getElementContaining` method works differently, it does not necessarily return an element whose one of the nodes correspond to the given point. Is this a bug or by intention? If this could be fixed first, the `evalPoint` fix will be straight forward.

**Comment by VikingScientist** [comment]
`LRSpline::LRSplineSurface::getElementContaining` returns the index to the element (not controlpoint) at the input parametric coordinates. You may then from this element extract all basis functions with support here using `LRSpline::Element::support` and finally fetch the indices or controlpoint-values to these basisfunctions.

However, note that the controlpoint values are not unique (this is true for tensor product splines as well). You may have multiple basisfunctions with the same controlpoint while still having a well-behaved (non-self-intersecting and non-degenerate) geometry. It is not sufficient to return an `int`, one needs to return an `IntVec`.

Also: the greville points of the basisfunctions (which could be used to identify controlpoint index) is not unique either for LR splines. In this case the tensor product version is more predictable in that they **are** unique.

In terms of creating a consistent API with the `ASMs2D` version, then it should be a simple change to create something along these lines

```c++
for(auto b : geo->getAllBasisfunctions() ) {
  for(int dim=0; dim<n_dimensions; dim++) {
    if( fabs(b->cp(dim) - target_control_point[dim]) < some_tolerance) {
      // store distance and keep this index
    }
  }
}
// return nearest index (or the first appearing one if multiple ones are available)
```

**Comment by kmokstad** [comment]
After #275, this functionality is no longer needed.
And since no straight-forward implementation we close this.
